src/bfs.jl

```diff
@@ -0,0 +1,111 @@
+"""
+Enumerates a context-free grammar up to a given depth in a breadth-first order.
+This means that smaller programs are returned before larger programs.
+"""
+mutable struct ContextFreeBFSEnumerator <: ExpressionIterator 
```

Comment:
should we make `ContextFreeBFSEnumerator` a subtype of `ContextFreeEnumerator` instead?

Reply:
That was what I initially did, but you cannot subtype concrete types in Julia unfortunately.